# Phase 1 - Loading Dataset

## Name: AYUSH KUMAR YADAV
## Registration Number: 23BDS0333
## Dataset: Lithium-Ion Battery Materials

### Objective
The objective of this notebook is to load the Lithium-Ion Battery Materials dataset
and perform an initial inspection of its structure. The dataset contains information
about material compositions, crystal structures, formation energy, energy above hull,
band gap, number of sites, density, volume, bandstructure availability, and crystal system.

The dataset is inspected using basic commands to understand its dimensions,
column names, data types, sample records, missing values, and duplicate records.

In [2]:
import pandas as pd

df = pd.read_csv("lithium_ion_23BDS0333.csv")

print(df.shape)
print(df.columns.tolist())
print(df.dtypes)
display(df.head())

(339, 11)
['Materials Id', 'Formula', 'Spacegroup', 'Formation Energy (eV)', 'E Above Hull (eV)', 'Band Gap (eV)', 'Nsites', 'Density (gm/cc)', 'Volume', 'Has Bandstructure', 'Crystal System']
Materials Id              object
Formula                   object
Spacegroup                object
Formation Energy (eV)    float64
E Above Hull (eV)         object
Band Gap (eV)            float64
Nsites                     int64
Density (gm/cc)          float64
Volume                   float64
Has Bandstructure           bool
Crystal System            object
dtype: object


,Materials Id,Formula,Spacegroup,Formation Energy (eV),E Above Hull (eV),Band Gap (eV),Nsites,Density (gm/cc),Volume,Has Bandstructure,Crystal System
0,mp-849394,Li2MnSiO4,Pc,-2.699,0.006,3.462,16,2.993,178.513,True,monoclinic
1,mp-783909,Li2MnSiO4,P21/c,-2.696,0.008,2.879,32,2.926,365.272,True,monoclinic
2,mp-761311,Li4MnSi2O7,Cc,-2.775,0.012,3.653,28,2.761,301.775,True,monoclinic
3,mp-761598,Li4Mn2Si3O10,C2/c,-2.783,0.013,3.015,38,2.908,436.183,True,monoclinic
4,mp-767709,Li2Mn3Si3O10,C2/c,-2.747,0.016,2.578,36,3.334,421.286,True,monoclinic


In [3]:
# ============================================================
# PHASE 1 - LOADING DATASET
# Name        : AYUSH KUMAR YADAV
# Registration: 23BDS0333
# Dataset     : Lithium-Ion Battery Materials
# ============================================================

import pandas as pd
import numpy as np

# ------------------------------------------------------------
# 1. LOAD DATASET
# ------------------------------------------------------------

file_path = "lithium_ion_23BDS0333.csv"

df = pd.read_csv(file_path)

print("=" * 70)
print("DATASET LOADED SUCCESSFULLY")
print("=" * 70)

# ------------------------------------------------------------
# 2. DISPLAY FIRST 5 RECORDS
# ------------------------------------------------------------

print("\nFIRST 5 RECORDS:")
display(df.head())

# ------------------------------------------------------------
# 3. DISPLAY LAST 5 RECORDS
# ------------------------------------------------------------

print("\nLAST 5 RECORDS:")
display(df.tail())

# ------------------------------------------------------------
# 4. DATASET DIMENSIONS
# ------------------------------------------------------------

print("\nDATASET DIMENSIONS:")
print("Number of Rows    :", df.shape[0])
print("Number of Columns :", df.shape[1])

# ------------------------------------------------------------
# 5. COLUMN NAMES
# ------------------------------------------------------------

print("\nCOLUMN NAMES:")
for i, column in enumerate(df.columns, start=1):
    print(f"{i}. {column}")

# ------------------------------------------------------------
# 6. DATA TYPES
# ------------------------------------------------------------

print("\nDATA TYPES:")
print(df.dtypes)

# ------------------------------------------------------------
# 7. DATASET INFORMATION
# ------------------------------------------------------------

print("\nDATASET INFORMATION:")
df.info()

# ------------------------------------------------------------
# 8. BASIC STATISTICAL SUMMARY
# ------------------------------------------------------------

print("\nNUMERICAL STATISTICAL SUMMARY:")
display(df.describe())

# ------------------------------------------------------------
# 9. COMPLETE STATISTICAL SUMMARY
# ------------------------------------------------------------

print("\nCOMPLETE DATASET SUMMARY:")
display(df.describe(include="all"))

# ------------------------------------------------------------
# 10. RANDOM SAMPLE
# ------------------------------------------------------------

print("\nRANDOM SAMPLE OF 5 RECORDS:")
display(df.sample(5, random_state=42))

# ------------------------------------------------------------
# 11. MISSING VALUES
# ------------------------------------------------------------

print("\nMISSING VALUES IN EACH COLUMN:")
missing_values = df.isnull().sum()
display(missing_values.to_frame("Missing Values"))

print("\nMISSING VALUE PERCENTAGE:")
missing_percentage = (df.isnull().sum() / len(df)) * 100
display(missing_percentage.to_frame("Missing Percentage"))

# ------------------------------------------------------------
# 12. DUPLICATE RECORDS
# ------------------------------------------------------------

print("\nNUMBER OF DUPLICATE ROWS:")
print(df.duplicated().sum())

# ------------------------------------------------------------
# 13. UNIQUE VALUES IN EACH COLUMN
# ------------------------------------------------------------

print("\nNUMBER OF UNIQUE VALUES IN EACH COLUMN:")

unique_values = df.nunique()
display(unique_values.to_frame("Unique Values"))

# ------------------------------------------------------------
# 14. DATASET MEMORY USAGE
# ------------------------------------------------------------

print("\nMEMORY USAGE:")
print(f"{df.memory_usage(deep=True).sum() / 1024:.2f} KB")

# ------------------------------------------------------------
# 15. FINAL DATASET SHAPE
# ------------------------------------------------------------

print("\nFINAL DATASET SHAPE:")
print(df.shape)

print("\n" + "=" * 70)
print("INITIAL DATASET INSPECTION COMPLETED")
print("=" * 70)

DATASET LOADED SUCCESSFULLY

FIRST 5 RECORDS:


,Materials Id,Formula,Spacegroup,Formation Energy (eV),E Above Hull (eV),Band Gap (eV),Nsites,Density (gm/cc),Volume,Has Bandstructure,Crystal System
0,mp-849394,Li2MnSiO4,Pc,-2.699,0.006,3.462,16,2.993,178.513,True,monoclinic
1,mp-783909,Li2MnSiO4,P21/c,-2.696,0.008,2.879,32,2.926,365.272,True,monoclinic
2,mp-761311,Li4MnSi2O7,Cc,-2.775,0.012,3.653,28,2.761,301.775,True,monoclinic
3,mp-761598,Li4Mn2Si3O10,C2/c,-2.783,0.013,3.015,38,2.908,436.183,True,monoclinic
4,mp-767709,Li2Mn3Si3O10,C2/c,-2.747,0.016,2.578,36,3.334,421.286,True,monoclinic



LAST 5 RECORDS:


,Materials Id,Formula,Spacegroup,Formation Energy (eV),E Above Hull (eV),Band Gap (eV),Nsites,Density (gm/cc),Volume,Has Bandstructure,Crystal System
334,mp-764961,Li6Co(SiO4)2,P1,-2.545,0.071,2.685,17,2.753,171.772,True,triclinic
335,mp-849520,LiCo3(SiO4)2,P1,-2.250,0.076,0.005,42,3.318,552.402,True,triclinic
336,mp-849656,Li5Co4(Si3O10)2,P1,-2.529,0.082,0.176,35,2.940,428.648,True,triclinic
337,mp-763557,LiCoSiO4,P1,-2.348,0.087,1.333,14,2.451,214.044,True,triclinic
338,mp-767320,Li3Co2(SiO4)2,P1,-2.406,0.09,0.323,15,3.043,176.207,False,triclinic



DATASET DIMENSIONS:
Number of Rows    : 339
Number of Columns : 11

COLUMN NAMES:
1. Materials Id
2. Formula
3. Spacegroup
4. Formation Energy (eV)
5. E Above Hull (eV)
6. Band Gap (eV)
7. Nsites
8. Density (gm/cc)
9. Volume
10. Has Bandstructure
11. Crystal System

DATA TYPES:
Materials Id              object
Formula                   object
Spacegroup                object
Formation Energy (eV)    float64
E Above Hull (eV)         object
Band Gap (eV)            float64
Nsites                     int64
Density (gm/cc)          float64
Volume                   float64
Has Bandstructure           bool
Crystal System            object
dtype: object

DATASET INFORMATION:
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 339 entries, 0 to 338
Data columns (total 11 columns):
 #   Column                 Non-Null Count  Dtype  
---  ------                 --------------  -----  
 0   Materials Id           339 non-null    object 
 1   Formula                338 non-null    object 
 2   Spa

,Formation Energy (eV),Band Gap (eV),Nsites,Density (gm/cc),Volume
count,339.000000,338.000000,339.000000,339.000000,339.000000
mean,-2.616950,2.082169,38.837758,2.984003,467.765619
std,0.183809,1.088661,23.133142,0.353968,292.674559
min,-2.985000,0.000000,10.000000,2.200000,122.581000
25%,-2.757500,1.266250,26.000000,2.760500,286.381500
50%,-2.605000,2.505500,31.000000,2.947000,358.537000
75%,-2.525500,2.968500,52.000000,3.106000,601.696500
max,-2.012000,3.823000,132.000000,4.201000,1518.850000



COMPLETE DATASET SUMMARY:


,Materials Id,Formula,Spacegroup,Formation Energy (eV),E Above Hull (eV),Band Gap (eV),Nsites,Density (gm/cc),Volume,Has Bandstructure,Crystal System
count,339,338,339,339.000000,339,338.000000,339.000000,339.000000,339.000000,339,338
unique,339,114,44,NaN,109,NaN,NaN,NaN,NaN,2,4
top,mp-849394,LiFeSiO4,P1,NaN,0.069,NaN,NaN,NaN,NaN,True,monoclinic
freq,1,42,72,NaN,9,NaN,NaN,NaN,NaN,274,137
mean,NaN,NaN,NaN,-2.616950,NaN,2.082169,38.837758,2.984003,467.765619,NaN,NaN
std,NaN,NaN,NaN,0.183809,NaN,1.088661,23.133142,0.353968,292.674559,NaN,NaN
min,NaN,NaN,NaN,-2.985000,NaN,0.000000,10.000000,2.200000,122.581000,NaN,NaN
25%,NaN,NaN,NaN,-2.757500,NaN,1.266250,26.000000,2.760500,286.381500,NaN,NaN
50%,NaN,NaN,NaN,-2.605000,NaN,2.505500,31.000000,2.947000,358.537000,NaN,NaN
75%,NaN,NaN,NaN,-2.525500,NaN,2.968500,52.000000,3.106000,601.696500,NaN,NaN



RANDOM SAMPLE OF 5 RECORDS:


,Materials Id,Formula,Spacegroup,Formation Energy (eV),E Above Hull (eV),Band Gap (eV),Nsites,Density (gm/cc),Volume,Has Bandstructure,Crystal System
284,mp-780147,LiMnSiO4,P1,-2.580,0.098,0.995,28,3.294,310.440,True,triclinic
116,mp-763255,LiCoSiO4,Pc,-2.372,0.064,1.373,14,2.964,176.961,True,monoclinic
113,mp-762978,LiCoSiO4,Cc,-2.376,0.06,1.281,112,2.998,1399.850,False,monoclinic
42,mp-761968,Li4FeSi2O7,Cc,-2.731,0.011,3.415,28,2.810,297.609,True,monoclinic
126,mp-764963,Li2CoSi3O8,P21,-2.772,0.085,3.159,28,2.771,341.661,True,monoclinic



MISSING VALUES IN EACH COLUMN:


,Missing Values
Materials Id,0
Formula,1
Spacegroup,0
Formation Energy (eV),0
E Above Hull (eV),0
Band Gap (eV),1
Nsites,0
Density (gm/cc),0
Volume,0
Has Bandstructure,0



MISSING VALUE PERCENTAGE:


,Missing Percentage
Materials Id,0.000000
Formula,0.294985
Spacegroup,0.000000
Formation Energy (eV),0.000000
E Above Hull (eV),0.000000
Band Gap (eV),0.294985
Nsites,0.000000
Density (gm/cc),0.000000
Volume,0.000000
Has Bandstructure,0.000000



NUMBER OF DUPLICATE ROWS:
0

NUMBER OF UNIQUE VALUES IN EACH COLUMN:


,Unique Values
Materials Id,339
Formula,114
Spacegroup,44
Formation Energy (eV),251
E Above Hull (eV),109
Band Gap (eV),306
Nsites,49
Density (gm/cc),300
Volume,339
Has Bandstructure,2



MEMORY USAGE:
107.47 KB

FINAL DATASET SHAPE:
(339, 11)

INITIAL DATASET INSPECTION COMPLETED


# Conclusion

The Lithium-Ion Battery Materials dataset was successfully loaded and inspected using Python.

The dataset contains 339 records and 11 attributes describing material properties and
crystal structures. The initial inspection identified numerical, categorical, and
Boolean variables. Basic statistical summaries, data types, missing values, duplicate
records, and unique-value counts were also examined.

This initial exploration provides a clear understanding of the dataset structure and
prepares the data for statistical analysis, missing-value handling, cleaning, and
transformation in the subsequent notebooks.